### PR TITLE
Docs update version and installation

### PR DESCRIPTION
**Pull request by haipinglu**
### Description

Update docs to have versions and installation instructions

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).

### CODE REVIEW

**Comment by bobturneruk** [comment]
When running:
```
PS C:\Users\bobturner\Documents\pykale\docs> make html
```
I get:
```
Running Sphinx v3.5.1

Extension error:
Could not import extension nbsphinx_link (exception: No module named 'nbsphinx_link')
make: *** [Makefile:20: html] Error 2
```
I'll investigate further.

**Comment by bobturneruk** [comment]
```
pip install nbsphinx_link
```
Seems to resolve the above.

**Comment by bobturneruk** [comment]
Perhaps I should run:
```
cd docs
pip install -r requirements.txt
```
Before running:
```
make html
```

File: docs/source/installation.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Installation
```
Comment (bobturneruk) [inline docs/source/installation.md:1]: This seems to duplicate content currently in readme.md - perhaps readme.md should be changed to link to this content so install instructions are all in one place?

**Comment by bobturneruk** [comment]
@haipinglu - sorry if I've missed this somewhere, but should the docs be build in the context of:
```
pip install -e .[dev]
```
or in the context of:
```
cd docs
pip install -r requirements.txt
```
please?

I guess it's fine to have the doc building requirements in the developer setup, but then maybe we don't need `requirements.txt` in `docs/`...

**Comment by haipinglu** [comment]
> ```
> pip install nbsphinx_link
> ```
> 
> Seems to resolve the above.

I will add it to `[dev]` option.

File: docs/source/installation.md (hunk 1 above)
Comment (haipinglu) [reply docs/source/installation.md:1]: > This seems to duplicate content currently in readme.md - perhaps readme.md should be changed to link to this content so install instructions are all in one place?

Great suggestion. I will keep a simple version in Readme and provide a link for more details.

**Comment by haipinglu** [comment]
> sorry if I've missed this somewhere, but should the docs be build in the context of:

Sorry for the confusion. The answer is actually both.

`requirements.txt` in `docs/` are the minimum required packages to build the docs. It is required by the [docs workflow](https://github.com/ammaraskar/sphinx-action#how-to-use) used, saying "If you have any Python dependencies that your project needs (themes, build tools, etc) then place them in a requirements.txt file inside your docs folder." It is also used by readthedocs.

We cannot remove it unless we reimplement the workflow. Now I have updated/synced `setup.py` so that you do not need to install extra packages if you installed pykale via `[dev]`, which is overcomplete for building `docs`.

**Comment by haipinglu** [comment]
@bobturneruk Many thanks for the helpful comments. I've updated further to address them. Please check and talk to you tomorrow.

**Comment by bobturneruk** [comment]
> We cannot remove it unless we reimplement the workflow. Now I have updated/synced setup.py so that you do not need to install extra packages if you installed pykale via [dev], which is overcomplete for building docs.

This kind of occurred to me after I'd commented - thanks for the explanation!

File: docs/requirements.txt [hunk 2]
```diff
@@ -5,15 +5,14 @@ nbsphinx
...
 
-ogb==1.2.0
 pillow>=6.2.0
 pytorch-lightning
 recommonmark
 scikit-learn
 sphinx-rtd-theme
 tensorly
-torch==1.7.0+cpu
+torch>=1.7.0
```
Comment (bobturneruk) [inline docs/requirements.txt:14]: Good spot!

File: README.md [hunk 3]
```diff
@@ -32,32 +32,13 @@
...
-To run the unit tests:
+Simple installation from [PyPI](https://pypi.org/project/pykale/):
 
 ```bash
-pytest
+pip install pykale
 ```
 
-More comprehensive test cases are not yet available.
+For more details and other options, please refer to [the installation guide](https://pykale.readthedocs.io/en/latest/installation.html).
```
Comment (bobturneruk) [inline README.md:41]: I like this.

**Comment by bobturneruk** [comment]
I haven't re-tested, but changes are small so please merge!

**Review by bobturneruk — APPROVED** [review]
Looks good!

**Pull request merged**